File: 0007/7.cpp
class Solution {
public:
    ListNode* reverseInteger(ListNode* head) {
        ListNode* prev=nullptr;
        ListNode* cur=head;
        while(cur){
            ListNode* next=cur->next;
            cur->next=prev;
            prev=cur;
            cur=next;
        }
        return prev;
    }
};

File: 0007/7.js
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseInteger = function(head) {
    let prev = null, cur = head;
    while (cur) {
        const next = cur.next;
        cur.next = prev;
        prev = cur;
        cur = next;
    }
    return prev;
};